src/ferc_xbrl_extractor/taxonomy.py

```diff
@@ -149,29 +161,35 @@ def from_list(cls, linkrole_list: List, concept_dict: ConceptDict) -> "LinkRole"
 
 
 class Taxonomy(BaseModel):
-    """
-    Pydantic model that defines an XBRL taxonomy.
+    """Pydantic model that defines an XBRL taxonomy.
 
-    A Taxonomy is an XBRL document, which is used to interpret/validate the facts in
-    individual filings.
+    XBRL Taxonomies are documents which are used to interpret facts reported in a
+    filing. Taxonomies are composed of linkroles that group concepts into fact
+    tables. This provides the structure and metadata that is used for extracting
+    XBRL data into a SQLite database.
     """
 
-    roles: List["LinkRole"]
+    roles: list[LinkRole]
 
     @classmethod
-    def from_path(cls, path: str, metadata_fname: str = ""):
-        """Construct taxonomy from taxonomy URL."""
+    def from_path(cls, path: str):
+        """Construct taxonomy from taxonomy URL.
+
+        Use Arelle to parse a taxonomy from a URL or local file path. The
+        structures Arelle uses to represent a taxonomy and its components are not
+        well documented and unintuitive, so the structures defined here are
+        instantiated and used instead.
+
+        Args:
+            path: URL or local path to taxonomy.
+        """
         taxonomy, view = load_taxonomy(path)
 
-        # Create dictionary mapping concept names to concepts (also strips prefix from name)
+        # Create dictionary mapping concept names to concepts
         concept_dict = {
             str(name): concept for name, concept in taxonomy.qnameConcepts.items()
         }
 
-        # References provide sometimes useful metadata that can be saved in JSON
-        if metadata_fname:
-            save_references(metadata_fname, concept_dict)
-
         roles = [
```

Comment:
Why was `save_references()` removed?

Reply:
At this point `save_references` was really an artifact of some earlier work that's not actually in use now. XBRL references contain additional metadata that I thought would be useful for linking XBRL data to DBF data, but it didn't end up working how I thought it might, and I figured at this point it would cause more confusion to leave this in.